### PR TITLE
netbsd.mkDerivation: fix darwin build

### PR DESCRIPTION
**Pull request by veprbl**
Fixes build of netbsd.make on darwin

Fixes: aab8c7ba437d ('netbsd: add cross target')
Resolves: #59141

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/os-specific/bsd/netbsd/default.nix [hunk 1]
```diff
@@ -51,7 +51,7 @@ let
 
     AR = "${stdenv'.cc.targetPrefix or ""}ar";
     CC = "${stdenv'.cc.targetPrefix or ""}cc";
-    CPP = if (stdenv'.cc.isClang or false) then "clang-cpp" else "cpp";
+    CPP = "${stdenv'.cc.targetPrefix or ""}cpp";
```
Comment (matthewbauer) [inline pkgs/os-specific/bsd/netbsd/default.nix]: @ericson2314 should cpp be prefixed? I feel like it is one case where we don't want the prefixed one. Everyone target could use the same cpp.

Comment (veprbl) [reply pkgs/os-specific/bsd/netbsd/default.nix]: I don't know much about cross stuff, but I think cpp knows about the target to be able to define macros like `__i386__` and such (e.g. https://github.com/openbsd/src/blob/master/libexec/tradcpp/config.h)

**Comment by matthewbauer** [comment]
Thanks for working on this!

File: pkgs/os-specific/bsd/netbsd/default.nix (hunk 1 above)
Comment (Ericson2314) [reply pkgs/os-specific/bsd/netbsd/default.nix]: It's just safer to prefix

**Pull request merged**